execution/executor/src/lib.rs

```diff
@@ -140,6 +147,7 @@ where
     /// Executes a block.
     pub fn execute_block(
         &self,
+        block_id: HashValue,
```

Comment:
I think the entire intent here is that executor isn't supposed to know about blocks. Is this intended to be parallelizable? Multiple traces concurrently? If not, could we just register this globally? If it is, perhaps we call this trace id, but that makes the APIa bit unfortunate....